src/GeekLearning.Storage.FileSystem/FileSystemStore.cs

```diff
@@ -130,21 +199,19 @@ private void EnsurePathExists(string path)
             }
         }
 
-        private Internal.FileSystemFileReference InternalGetAsync(IPrivateFileReference file)
+        private static string GenerateEtag(string fileSystemPath)
         {
-            var reference = InternalGetOrCreateAsync(file);
-            if (File.Exists(reference.FileSystemPath))
+            var data = File.ReadAllBytes(fileSystemPath);
```

Comment:
I remembered I had the same issue here : https://github.com/geeklearningio/gl-dotnet-storage/blob/develop/src/GeekLearning.Storage.Azure/Internal/AzureFileReference.cs#L68